packages/nimble-components/src/rich-text/models/tests/markdown-parser.spec.ts

```diff
@@ -1243,14 +1331,73 @@ describe('Markdown parser', () => {
                 ]
             );
 
-            expect(getTagsFromElement(doc)).toEqual([
+            expect(getTagsFromElement(parserContent.fragment)).toEqual([
                 'P',
                 `${anchorTag}`.toUpperCase()
             ]);
-            expect(getLastChildElementAttribute('href', doc)).toBe(
+            expect(
+                getLastChildElementAttribute('href', parserContent.fragment)
+            ).toBe('https://user/1');
+            expect(getLeafContentsFromElement(parserContent.fragment)).toEqual([
                 'https://user/1'
+            ]);
+            expect(parserContent.mentionedHrefs).toEqual([]);
+        });
+
+        it('should return markdownParserMentionConfig matching mentioned href list with valid mention markdown', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'username1' },
+                    { key: 'user:2', displayName: 'username2' }
+                ],
+                '^user:(.*)'
+            ));
+            await connect();
+            const parserContent = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<user:1234-5678> <user:135>',
+                [
+                    new MarkdownParserMentionConfiguration(
+                        element.mentionInternals
+                    )
+                ]
             );
-            expect(getLeafContentsFromElement(doc)).toEqual(['https://user/1']);
+            expect(parserContent.mentionedHrefs).toEqual([
+                'user:1234-5678',
+                'user:135'
+            ]);
+        });
+
+        it('should return empty mentioned href list with invalid mention markdown', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'username1' },
+                    { key: 'user:2', displayName: 'username2' }
+                ],
+                '^user:(.*)'
+            ));
+            await connect();
+            const parserContent = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<invalid:1234-5678>',
+                [
+                    new MarkdownParserMentionConfiguration(
+                        element.mentionInternals
+                    )
+                ]
+            );
+            expect(parserContent.mentionedHrefs).toEqual([]);
+        });
+
+        it('should return empty mentioned href list called without markdownParserMentionConfig', async () => {
```

Comment:
This test seems weird to me: it's not actually using the element that it creates. If we want to test the parser standalone then you could delete the element code. If we want it to be an integration test following the same pattern as the other tests in the file then we should instead configure the element to have no mention config.

Reply:
Removed the element code.